mirrord/config/src/lib.rs

```diff
@@ -278,12 +278,24 @@ pub struct LayerConfig {
 }
 
 impl LayerConfig {
-    pub fn from_env() -> Result<Self, ConfigError> {
+    /// Generate a config from the environment variables and/or a config file.
+    /// On success, returns the config and a vec of warnings.
+    /// To be used from CLI to verify config and print warnings
+    pub fn from_env_with_warnings() -> Result<(Self, Vec<String>), ConfigError> {
+        let mut warnings = Vec::new();
```

Comment:
For consistency with other APIs and `from_env` prettier
```suggestion
    pub fn from_env_with_warnings(warnings: &mut Vec<String>) -> Result<(Self, Vec<String>), ConfigError> {
```

Reply:
I think the internal API shouldn't be consistent with external tbh.. also makes funky code from caller.